sn_transfers/README.md

```diff
@@ -1,17 +1,317 @@
 
-`Transfer`s are directed to `UniquePubKey`s, which are derived from `MainPubKey`s. Using a `DerivationIndex`, the recipient can generate the necessary `SecretKey` to spend the `NanoTokens`.
+Unlike one might expect, the `MainPubkey` itself never owns any money: `UniquePubkey`s derived from it do. Value is owned by those `UniquePubkey`s which are spendable only once in the form of a `Spend` uploaded at that `UniquePubkey`'s address (known as a `SpendAddress`) on the Network. 
 
-Since most of the required information is stored in a `SignedSpend` on the network, we also provide a [`CashNoteRedemption`](https://github.com/maidsafe/sn_transfers/blob/main/src/cash_note_redemption.rs) struct. This struct contains the minimum information needed to retrieve a full `Spend` from the network and generate the keys required to spend the `NanoTokens`.
+The way we obtain those `UniquePubkey`s is by using bls key derivation, an algorithm which creates a new key from another key by using a large number called a `DerivationIndex`. `UniquePubkey`s are derived from the `MainPubkey`. To spend the value owned by a `UniquePubkey`, one uses the associated `DerivedSecretKey` which was derived from the `MainSecretKey` using the same `DerivationIndex` as was used to create the `UniquePubkey`.
 
-For error handling, we expose [`Error`](https://github.com/maidsafe/sn_transfers/blob/main/src/error.rs) and [`Result`](https://github.com/maidsafe/sn_transfers/blob/main/src/result.rs) types.
+This `DerivedSecretKey` is used to sign the `Spend` which is then sent to the Network for validation and storage. Once the Network has stored and properly replicated that `Spend`, that `UniquePubkey` is considered to be spent and cannot ever be spent again. If more than one `Spend` entry exist at a given `SpendAddress` on the Network, that key is considered to be burnt which makes any `Spend` refering to it unspendable. 
+
+Without the `DerivationIndex`, there is no way to link a `MainPubkey` to a `UniquePubkey`. Since `UniquePubkey`s are spendable only once, this means every transaction involves new and unique keys which are all unrelated and unlinkable to their original owner's `MainPubkey`.
+
+Under the hood, those types are simply:
+
+- `MainPubkey` => `blsttc::PublicKey`
+- `UniquePubkey` => `blsttc::PublicKey` (derived from `MainPubkey`)
+- `MainSecretKey` => `blsttc::SecretKey`
+- `DerivedSecretKey` => `blsttc::SecretKey` (derived from `MainSecretKey`)
+- `DerivationIndex` => `u256` (big number impossible to guess, used to derive keys)
+
+
+## Spends
+
+When a `UniquePubkey` is spent, the owner creates a `Spend` and signs it with the associated `DerivedSecretKey` before uploading it to the Network. A `Spend` contains the following information:
+
+```rust
+pub struct Spend {
+    pub unique_pubkey: UniquePubkey,
+    pub ancestors: BTreeSet<UniquePubkey>,
+    pub descendants: BTreeMap<UniquePubkey, NanoTokens>,
```

Comment:
still need to keep the `royalties` member field

also, shall the doc cover the `reason` member field as well?

if this doc is just about currency & spend DAG, which these two fields (especially the reason field) are not that related,
at least put a `...` below the line of `descendants`, to avoid give a mis-impression to the readers that `Spend` struct only has that three member fields ?

Reply:
Yes I didn't document that as those fields are more likely to evolve thus the `...`

Reply:
We could add those once stabilized?